### PR TITLE
bpo-42340: Document issues around KeyboardInterrupt

### PR DESCRIPTION
**Pull request by benfogle**
Update documentation to note that in some circumstances,
KeyboardInterrupt may cause code to enter an inconsistent state. Also
document sample workaround to avoid KeyboardInterrupt, if needed.

https://bugs.python.org/issue42340

### CODE REVIEW

File: Doc/library/exceptions.rst
Comment (birkenfeld) [inline Doc/library/exceptions.rst]: This needs clarification that it only applies in cases where `KeyboardInterrupt` is caught.  Otherwise people will become afraid to press Ctrl-C...

I'd also directly mention what to do: "... you should not try to catch KeyboardInterrupt and continue, but let it end the program as quickly as possible, or avoid raising it completely (see ...)"

**Review by birkenfeld — DISMISSED** [review]

File: Doc/library/exceptions.rst
Comment (benfogle) [reply Doc/library/exceptions.rst]: I have updated the pull request.

File: Doc/library/exceptions.rst
Comment (birkenfeld) [inline Doc/library/exceptions.rst]: two "may"s here... otherwise looks good to me.

Comment (benfogle) [reply Doc/library/exceptions.rst]: whoops. Thanks.

File: Doc/library/signal.rst
Comment (JelleZijlstra) [inline Doc/library/signal.rst]: Ctrl+C isn't appropriate terminology for Windows. Maybe we should just say KeyboardInterrupt?

File: Doc/library/signal.rst
Comment (JelleZijlstra) [inline Doc/library/signal.rst:691]: The example in your issue (KeyboardInterrupt in the middle of an `__enter__` was very helpful for me in understanding the problem. What do you think of adding it here?

**Review by JelleZijlstra — CHANGES_REQUESTED** [review]
Thanks for your contribution, and sorry it's been waiting for so long! I'm happy to help merge this but have a few comments.

File: Doc/library/signal.rst
Comment (benfogle) [reply Doc/library/signal.rst]: fixed

File: Doc/library/signal.rst
Comment (benfogle) [reply Doc/library/signal.rst:691]: I added a simple example after this paragraph.

**Review by JelleZijlstra — APPROVED** [review]

**Comment by JelleZijlstra** [comment]
Thanks! I'll merge this in a few days unless another core dev comes along and has concerns.

**Review by asvetlov — APPROVED** [review]

**Pull request merged**
